internal/printer/printer_test.go

```diff
@@ -371,7 +372,7 @@ const someProps = {
 
 // Full Astro Component Syntax:
 // https://docs.astro.build/core-concepts/astro-components/`},
-				styles: []string{":root{font-family:system-ui;padding:2em 0;}.counter{display:grid;grid-template-columns:repeat(3,minmax(0,1fr));place-items:center;font-size:2em;margin-top:2em;}.children{display:grid;place-items:center;margin-bottom:2em;}"},
+				styles: []string{fmt.Sprintf(`{props:{"data-astro-id":"HMNNHVCQ"},children:%s:root{font-family:system-ui;padding:2em 0;}.counter{display:grid;grid-template-columns:repeat(3,minmax(0,1fr));place-items:center;font-size:2em;margin-top:2em;}.children{display:grid;place-items:center;margin-bottom:2em;}%s}`, BACKTICK, BACKTICK)},
```

Comment:
This test shows a big change: we're passing `{props:{},children`code`}` to `$$results.styles` and `$$results.scripts`! This allows us to do hoisting without losing the attributes.

Reply:
nit: not crazy about `fmt.Sprintf()` in a test, but it’s fine for this PR. We probably need to split up some of these tests better anyway in the future (not this PR)